src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,792 @@
+                        symbolInfo.Symbol is IMethodSymbol methodSymbol &&
+                        methodSymbol.Parameters.Length == 1)
+                    {
+                        processingState.AddVariableName(queryContinuation.Identifier.ValueText);
+                        processingState.AddStatementProcessingMethod(
+                            s => AddToBlockTop(
+                                 SyntaxFactory.LocalDeclarationStatement(
+                                    SyntaxFactory.VariableDeclaration(
+                                        VarNameIdentifier,
+                                        SyntaxFactory.SingletonSeparatedList(
+                                            SyntaxFactory.VariableDeclarator(
+                                                queryContinuation.Identifier,
+                                                argumentList: null,
+                                                SyntaxFactory.EqualsValueClause(selectClause.Expression))))), s));
+
+                        return TryProcessQueryBody(queryContinuation.Body, processingState);
+                    }
+                }
+
+                return false;
+            }
+
+            private bool TryProcessWhereClause(WhereClauseSyntax whereClause, ProcessingState processingState)
+            {
+                if (IsValidWhereClauseSymbol(whereClause))
+                {
+                    processingState.AddStatementProcessingMethod(
+                         s => SyntaxFactory.Block(
+                             SyntaxFactory.IfStatement(
+                                 ProcessCondition((whereClause).Condition),
+                                 s)));
+
+                    return true;
+                }
+
+                return false;
+            }
+
+            private bool IsValidWhereClauseSymbol(WhereClauseSyntax whereClause)
+            {
+                if (!TryGetMethodSymbol(whereClause, out var methodSymbol))
+                {
+                    return false;
+                }
+
+                var returnType = methodSymbol.ReturnType;
+                if (returnType.OriginalDefinition?.SpecialType != SpecialType.System_Collections_Generic_IEnumerable_T)
+                {
+                    return false;
+                }
+
+                var typeArguments = returnType.GetTypeArguments();
+                if (typeArguments.Length != 1)
+                {
+                    return false;
+                }
+
+                var typeArgument = typeArguments.Single();
+                switch (methodSymbol.Parameters.Length)
+                {
+                    // IEnumerable<T> Where(Func<T, bool> predicate) 
+                    case 1:
+                        return IsPredicate(methodSymbol.Parameters.First().Type, typeArgument);
+
+                    // IEnumerable<T> Where(this IEnumerable<T>, Func<T, bool> predicate) 
+                    case 2:
+                        return
```

Comment:
>return [](start = 24, length = 6)

Is this code even reachable? I assume SemanticModel always returns reduced form for extension methods. When would we have a method symbol with two parameters here?